### ISSUE TITLE
additional/missing ags after flows_timexagsxtech() #26

### ISSUE THREAD

**Issue by nailend** [windnode/WindNODE_ABW#38]
#26 

1. [x] Additional 7-digit AGS appear after [flows_timexagsxtech()](https://github.com/windnode/WindNODE_ABW/blob/f63c8b701f0e4275c09e5e3ee98045849b92e452/windnode_abw/analysis/tools.py#L477)

- flows_timexagsxtech['Stromerzeugung'] list 24 unique AGS
- flows_timexagsxtech['Stromnachfrage'] only list 20 unique AGS
- Additional 7-digit AGS: {'1500100', '1508201', '1508218', '1509137}
- [fixed](https://github.com/windnode/WindNODE_ABW/commit/16e804814266f3121f79d56d2281f6cef655d83f)

2. [x] "Stromexport" and "Stromimport" only list 8 unique AGS @nesnoj 
- shouldnt there be an entry even if the value is zero?
- is export/import only possible if there is a substation to a higher gridlevel?
- is this plausible?

3. [x] atypical line_ids in "Stromnetz" @nesnoj 
- the majority of lines only have a 5-digit-ID
- atypical ID's: '461202177', '461202186', '99999999990', '99999999991' 
- are these special ones?

**Comment by nesnoj** [comment]
> #26
> 
> 1. [x]  Some additional 7-digit AGS appear after [flows_timexagsxtech()](https://github.com/windnode/WindNODE_ABW/blob/f63c8b701f0e4275c09e5e3ee98045849b92e452/windnode_abw/analysis/tools.py#L477)

First, I renamed the variable as the var + called function [had the same name](https://github.com/windnode/WindNODE_ABW/blob/16e804814266f3121f79d56d2281f6cef655d83f/windnode_abw/run_analysis.py#L36) which is generally a very bad idea (b663b9f).
I did not see any 7-digit AGS in the results, if I do
```
for k, v in flows_txaxt.items():
    for mi_level in [0,1]:
        column_name = v.index.names[mi_level]
        if column_name == 'ags':
            ags_list = list(v.index.get_level_values(mi_level).unique())
            print(f'{len(ags_list)} {column_name} in {k}: {ags_list}')
```
to see all ags of all results, I get
```
20 ags in Stromerzeugung: ['15001000', '15082005', ...]
20 ags in Stromnachfrage: ['15001000', '15082005', ...]
8 ags in Stromexport: ['15082340', '15082256', ...]
8 ags in Stromimport: ['15082340', '15082256', ...]
```
which looks ok.
I guess u fixed that one already in 16e8048?

> 
> * flows_timexagsxtech['Stromerzeugung'] list 24 unique AGS
> * flows_timexagsxtech['Stromnachfrage'] only list 20 unique AGS
> * These are the additional {'1500100', '1508201', '1508218', '1509137'} with only 7 digits
> 
> 1. [x]  "Stromexport" and "Stromimport" only list 8 unique AGS @nesnoj

Yes, that's ok as only 8 muns do have connections to the national grid on 110kV level.

> 2. [x]  strange line_ids in "Stromnetz" @nesnoj
> 
> * '461202177', '461202186', '99999999990', '99999999991']

What's strange with them? Those IDs come from the open_eGo and we just kept the IDs..

**Comment by nesnoj** [comment]
> #26
> 
> 1. [x]  Additional 7-digit AGS appear after [flows_timexagsxtech()](https://github.com/windnode/WindNODE_ABW/blob/f63c8b701f0e4275c09e5e3ee98045849b92e452/windnode_abw/analysis/tools.py#L477)
> 
> * flows_timexagsxtech['Stromerzeugung'] list 24 unique AGS
> * flows_timexagsxtech['Stromnachfrage'] only list 20 unique AGS
> * Additional 7-digit AGS: {'1500100', '1508201', '1508218', '1509137}
> * [fixed](https://github.com/windnode/WindNODE_ABW/commit/16e804814266f3121f79d56d2281f6cef655d83f)

ah ok

> 
> 1. [ ]  "Stromexport" and "Stromimport" only list 8 unique AGS @nesnoj
> 
> * shouldnt there be an entry even if the value is zero?

no

> * is export/import only possible if there is a substation to a higher gridlevel?

no, always when there's a connection to a non-region bus (applies for the 8 AGS you mentioned)

> * is this plausible?

yepp

> 1. [ ]  atypical line_ids in "Stromnetz" @nesnoj
> 
> * the majority of lines only have a 5-digit-ID
> * atypical ID's: '461202177', '461202186', '99999999990', '99999999991'
> * are these special ones?

kind of (a later manual update), but you don't have to worry about it for the results...
